kore/src/Kore/Logger/SMTCondition.hs

```diff
@@ -0,0 +1,49 @@
+{- |
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+
+-}
+
+module Kore.Logger.SMTCondition
+    ( SMTCondition (..)
+    , logSMTCondition
+    ) where
+
+import Kore.Logger
+    ( Entry (..)
+    , Severity (Info)
+    , MonadLog
+    , logM
+    )
+import Kore.Internal.Predicate
+    ( Predicate
+    )
+import Kore.Internal.TermLike
+import Data.Text.Prettyprint.Doc
+    ( Pretty (..)
+    )
+import Data.Typeable
+import Kore.Unparser
+
+newtype SMTCondition variable =
```

Comment:
Please name this `InfoEvaluateCondition`. It must not have any type parameters. The type of the condition should be `Predicate Variable`; this should prevent all the extra constraints.